MetaMorpheus/EngineLayer/GlycoSearch/GlycoSearchEngine.cs

```diff
@@ -710,6 +708,56 @@ private List<GlycoSpectralMatch> Find_N_O_Glycopeptide(Ms2ScanWithSpecificMass t
             return possibleMatches;
         }
 
+        /// <summary>
+        /// Valid the Graph created by this modPos and glycanBox.
+        /// Check if the motif in peptide is sufficient to cover the motif in glycanBox.
+        /// </summary>
+        /// <param name="modPos"></param>
+        /// <param name="glycanBox"></param>
+        /// <returns></returns>
+        private bool GraphCheck(Dictionary<int, string> modPos, GlycanBox glycanBox)
+        {
+            // If the motifs number is less than the glycanBox, we can skip this graph.
+            if (modPos.Count < glycanBox.NumberOfMods)
+                return false;
+
+            // Calculate the motif in glycanBox.
+            var motifInBox = new Dictionary<string, int>();
+            foreach (var modId in glycanBox.ModIds)
+            {
+                var motif = GlycanBox.GlobalOGlycans[modId].Target.ToString();
+
+                if (!motifInBox.ContainsKey(motif))
+                {
+                    motifInBox[motif] = 0;
+                }
+                motifInBox[motif]++;
+            }
+
+            // Calculate the motif in peptide.
+            var motifInPeptide = new Dictionary<string, int>();
+            for (int i = 0; i <= modPos.Count-1; i++)
+            {
+                var motif = modPos.Values.ElementAt(i);
```

Comment:
You should just use a foreach loop. Dictionaries aren't ordered, so there's no guarantee you'll access every element doing it this way.